zrml/market-commons/src/migrations.rs

```diff
@@ -14,3 +14,260 @@
+                    dispute_mechanism: legacy_market.dispute_mechanism,
+                    deadlines,
+                };
+                put_storage_value::<Option<MarketOf<T>>>(
+                    MARKET_COMMONS,
+                    MARKETS,
+                    &key,
+                    Some(new_market),
+                );
+                total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            }
+        }
+        log::info!("Completed updates of markets");
+        StorageVersion::new(MARKET_COMMONS_NEXT_STORAGE_VERSION).put::<Pallet<T>>();
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        let deadlines = Deadlines {
+            oracle_delay: BLOCKS_PER_DAY.saturated_into::<u32>().into(),
+            oracle_duration: BLOCKS_PER_DAY.saturated_into::<u32>().into(),
+            dispute_duration: BLOCKS_PER_DAY.saturated_into::<u32>().into(),
+        };
+        for (market_id, market) in storage_iter::<Option<MarketOf<T>>>(MARKET_COMMONS, MARKETS) {
+            if let Some(market) = market {
+                assert_eq!(
+                    market.deadlines, deadlines,
+                    "found unexpected deadlines in market. market_id: {:?}, deadlines: {:?}",
+                    market_id, market.deadlines
+                );
+            }
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::mock::{ExtBuilder, Runtime};
+    use core::fmt::Debug;
+    use frame_support::{storage::migration::get_storage_value, Blake2_128Concat, StorageHasher};
+    use parity_scale_codec::Encode;
+
+    type MarketId = <Runtime as Config>::MarketId;
+
+    #[test]
+    fn test_on_runtime_upgrade() {
+        ExtBuilder::default().build().execute_with(|| {
+            StorageVersion::new(MARKET_COMMONS_REQUIRED_STORAGE_VERSION).put::<Pallet<Runtime>>();
+            let (legacy_markets, expected_markets) = create_test_data();
+            populate_test_data::<Blake2_128Concat, MarketId, Option<LegacyMarketOf<Runtime>>>(
+                MARKET_COMMONS,
+                MARKETS,
+                legacy_markets,
+            );
+            UpdateMarketsForDeadlines::<Runtime>::on_runtime_upgrade();
+            for (market_id, market_expected) in expected_markets.iter().enumerate() {
+                let key_hash = key_to_hash::<Blake2_128Concat, MarketId>(market_id);
+                let market_actual = get_storage_value::<Option<MarketOf<Runtime>>>(
+                    MARKET_COMMONS,
+                    MARKETS,
+                    &key_hash,
+                )
+                .unwrap()
+                .unwrap(); // Unwrap twice because get_storage returns an option!
+                assert_eq!(market_actual, *market_expected);
+            }
```

Comment:
The problem with this test is that you're writing to storage manually and then later reading from storage manually, meaning that you don't check if or if not you're using the correct encoding.

```suggestion
            for (market_id, market_expected) in expected_markets.iter().enumerate() {
                let market_actual = Pallet::<Runtime>::market(&(market_id as u128)).unwrap();
                assert_eq!(market_actual, *market_expected);
            }
```